### DISCUSSION TITLE
[Idea] Make it possible to activate a language in BO only

### DISCUSSION THREAD

**Discussion by matdu58** [PrestaShop/PrestaShop#32275] [Ideas]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Observed problem

When we add several languages, but do not activate them, it's impossible to translate product title / description / add photos for that specific language...
It was working on PS 1.6, but not on PS 1.7
In fact, we don't need to use languages in the website front-office (just need to leave French activated), but we need to make translations in the back-office only (in order to send product data to some foreign marketplaces, in the right language).

### Expected behavior

It should be possible to disable a language in FO only.

When editing a language in BO, in International > Localization > Languages, there should be an additional switch below "Status" called "Display in Front Office".

This switch can only be activated when "Status" is enabled (activating Status activates FO automatically). Conversely, toggling Status to disabled should automatically disable "Display in Front Office" as well.

These are the only accepted states:

Description | Status | Display in FO
------------|--------|--------------
Disabled everywhere | Off | Off
Enabled everywhere | On | On
Enabled for BO only | On | Off

Disabling "Display in Front Office" will prevent users from choosing said language in the dropdown menu in FO. Customers who had chosen the disabled language before it became disabled will continue to see the FO in that language.

**Comment by hibatallahAouadni** [comment]
Hello @matdu58 

> Go to any product page, you can see only active languages (French for example) is available, Spanish is not available to translate product title / description / add photos for Spanish...

This is the expected behavior, if you want to translate into Spanish you have to activate it.
Let's check with @PrestaShop/product-team wdyt?

Thanks!

**Comment by matdu58** [comment]
Hello @hibatallahAouadni

No, you don't understand...
In fact, I don't need to use Spanish in the front-office (only French have to be activated), but I just need to manage the Spanish language (make product translations) only in the back-office (then to export product descriptions on the marketplaces for example). 

That was possible in PS 1.6, but not in PS 1.7

**Comment by MatShir** [comment]
I guess the ideal, it would be to have 2 states for a language: 
- Enable/Disable for the BO
- Enable/Disable for the FO

**Comment by matdu58** [comment]
@hibatallahAouadni 
You sent a message : "it works now on 1.7.8.1 you can edit you language no need to enable it on front office", but it seems it has been removed. Are you sure that on 1.7.8.1 it's OK now ?

@MatShir 
Yes, that can be a great idea, and easy to implement.

**Comment by go-sudo** [comment]
Hello
We can now see language option on modules even when it disabled but not yet on products

**Comment by hibatallahAouadni** [comment]
Hello @matdu58 

> You sent a message : "it works now on 1.7.8.1 you can edit you language no need to enable it on front office", but it seems it has been removed. Are you sure that on 1.7.8.1 it's OK now ?

No, I didn't send any message :open_mouth:

**Comment by matdu58** [comment]
@go-sudo 
Yes, of course, all translations are available for modules, even if language is disabled.
And I also need that option availble for products too.
Will Prestashop add this feature ?

@hibatallahAouadni 
Strange, I got such message from you in my mailbox...

**Comment by hibatallahAouadni** [comment]
> @hibatallahAouadni
Strange, I got such message from you in my mailbox...

What!!?? me? :open_mouth:

**Comment by MatShir** [comment]
Your feature is validated by the team. But it won't be prioritized for the next version but however, PrestaShop is an open-source project, if you have the technical knowledge or find someone that could don't hesitate to propose a pull request ([find more information here](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/)). We will be happy to review your contribution to the PrestaShop core.

**Comment by matdu58** [comment]
> it works now on 1.7.8.1 you can edit you language no need to enable it on front offic

Sorry, it was Nami Abdelmoula who wrote that, not you. Sorry for the confusion. But it has been removed.

**Comment by matdu58** [comment]
> Your feature is validated by the team. But it won't be prioritized for the next version but however, PrestaShop is an open-source project, if you have the technical knowledge or find someone that could don't hesitate to propose a pull request ([find more information here](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/)). We will be happy to review your contribution to the PrestaShop core.

@MatShir 
Great news, do you know what time it can takes ?
Unfortunately, I don't have knowledge in development...
If someone can do that, that would be wonderful.

**Comment by eternoendless** [comment]
> I guess the ideal, it would be to have 2 states for a language:
> 
> * Enable/Disable for the BO
> * Enable/Disable for the FO

@MatShir in what scenario you would want to activate the language in FO but not in BO?

IMO we should keep the master "enable" switch (which disables everywhere), and a "enable in FO" switch that can only be activated when the other one is active as well. Needless to say that if the language is fully disabled it shouldn't appear in the BO.

**Comment by matdu58** [comment]
@eternoendless Yes, that should be a good idea !
As far as I am concerned, I really need ASAP to use the languages in BO, but only 1 language (French) in FO

**Comment by eternoendless** [comment]
I have updated the issue description with the specifications

**Comment by eternoendless** [comment]
> As far as I am concerned, I really need ASAP to use the languages in BO

The Product Team will take it into consideration for future developments, but please be aware that there is no guarantee that this feature will be developed anytime soon. If this is important to you, we strongly suggest you consider [hiring a specialist](https://www.prestashop.com/en/experts) to help you create it. 

If you or anyone else is willing to implement this feature into PrestaShop as a [Pull Request](https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/), the teams will be happy to help by ensuring that it is appropriate and coherent with the rest of the project. If that is the case, please let us know!

**Comment by matdu58** [comment]
Hello !
To keep you updated, I found an alternative until a solution will be added natively to Prestashop.
That is a new module available on Addons here : 
https://addons.prestashop.com/fr/international-localisation/86918-bloc-de-choix-de-langue-personnalise.html
From your side, tell me when the native solution will have been implemented, so that I can remove the module (I will use it temporarily).

**Comment by Hlavtox** [comment]
Hi everyone, I think that this is very specific behavior and nobody except @matdu58 is going to need this. Why we should have two language-enable switches?

If you disable a language, you should see it in your backoffice, so you can work on it. No need to have any other switch.

**Comment by matdu58** [comment]
Hi everyone,

No, because actually with PS 1.7.7.4, when you activate a new language (example Spanish) in addition to the default language (French in my case), then you deactivate it (I don't want to display Spanish in the front office, but only French), you can not edit products informations (titles, descriptions) for the other languages (Spanish). 

It was possible in PS 1.6.X
So I project to buy that module on Addons can solve that problem temporarily : 
https://addons.prestashop.com/en/international-localization/86918-turn-off-languages-without-losing-control.html

**Comment by eternoendless** [comment]
> If you disable a language, you should see it in your backoffice, so you can work on it. No need to have any other switch.

This makes sense, but I believe the "status" behavior right now works as a soft delete, rendering the language disabled everywhere. Also there's a use case where you want the language enabled in the BO for the employee, but you don't want it showing up in the FO, nor you want to have it polluting your product data.

**Comment by Hlavtox** [comment]
@eternoendless The status switch just disables it in FO, but you can use the language in BO just fine.

There are some places in BO where it did not work - old product page for example, but this will be fixed now on v2 product page.

**Comment by matdu58** [comment]
@Hlavtox @eternoendless Below the detailled issue :

French language is enabled by default, that means all products are automatically translated into French.

Pictures 1 and 2 : 
I activate Italian and Spanish languages, so the products can be translated into these 2 languages. But the problem is that on the FO, languages are also activated. Consequently, my website is now "mysite.com/fr" (and also "mysite.com/it" + "mysite.com/es") instead of only "mysite.com" (no need "/fr" for only French, the website will never be translated into another language). I don't want that, but only French for BO.

Pictures 3 and 4 : 
I deactivate Italian and Spanish languages, so the products can be translated only in French. But the problem is that I want also to translate in Italian and Spanish. Actually, this is not possible.

**Will that improvement be implemented soon, or must I buy the module I mentionned before ?**

Picture 1 : 
![Capture 1](https://user-images.githubusercontent.com/94061573/169657798-1b16a2ac-537d-480a-8a53-24d100aef3c5.png)
Picture 2 : 
![Capture 2](https://user-images.githubusercontent.com/94061573/169657799-a3fbc302-399f-4644-8b39-a94a9320dba2.png)
Picture 3 : 
![Capture 3](https://user-images.githubusercontent.com/94061573/169657800-945568f0-16d8-4aa2-9c9a-86c191d0ea84.png)
Picture 4 : 
![Capture 4](https://user-images.githubusercontent.com/94061573/169657802-83c33632-58d3-46d1-b30b-1a314f578716.png)

**Comment by Hlavtox** [comment]
@matdu58 This is fixed on a new product page coming in Prestashop 8. 😉

**Comment by matdu58** [comment]
@Hlavtox Great ! You mean Prestashop 1.8 ? When will that new version be released approximately ?

**Comment by Hlavtox** [comment]
@matdu58 It's gonna be called Prestashop 8, we drop the 1.X versioning for easier orientation. But the upgrade will be as smooth as 1.7.6 to 1.7.7, don't worry. 

Details and approximate dates here: https://github.com/PrestaShop/PrestaShop/issues/26427

**Comment by matdu58** [comment]
@Hlavtox OK, nice ! Do you have any idea of the release of that new version ? To see if I can wait or if I have to buy the previous module I mentionned before...

**Comment by eternoendless** [comment]
I suggest not to wait for the new version. I don't think it will be available before September.
